### PR TITLE
refactor(lsp): changes for lsp_types 0.97.0

### PR DESCRIPTION
**Pull request by nayeemrmn**
~~Blocked~~ by https://github.com/ebkalderon/tower-lsp/pull/401. But currently this PR patches that revision in `Cargo.toml` so this could be landed.

Unblocks jupyter cell continuity: https://github.com/denoland/vscode_deno/issues/932.

### CODE REVIEW

File: Cargo.toml
Comment (bartlomieju) [inline Cargo.toml:49]: Could you move it to the very bottom? When we're trying `deno_core` locally `[patch]` is usually placed at the bottom

File: cli/lsp/capabilities.rs
Comment (bartlomieju) [inline cli/lsp/capabilities.rs:158]: I assume this is gonna happen in the follow up PR?

**Review by bartlomieju — COMMENTED** [review]
Looks reasonable, I'll wait for David's approval

File: cli/lsp/capabilities.rs
Comment (nayeemrmn) [reply cli/lsp/capabilities.rs]: Yeah, though I hadn't decided that when writing that. Removed the comment.

File: Cargo.toml
Comment (dsherret) [inline Cargo.toml:401]: We can't publish to crates.io with this. Are we just not going to publish with this PR for now?

**Review by dsherret — APPROVED** [review]
LGTM, but we can't publish to crates.io after landing this pr (which may or may not be a problem)

File: Cargo.toml
Comment (bartlomieju) [reply Cargo.toml:401]: Why can't we publish to crates.io with this one?

Comment (dsherret) [reply Cargo.toml:401]: IIRC it will error because all the dependencies aren't on crates.io (and I would really hope it has this behaviour)

Comment (bartlomieju) [reply Cargo.toml:401]: Got it, then we should publish our fork to crates.io

Comment (bartlomieju) [reply Cargo.toml:401]: Published as https://crates.io/crates/deno_tower_lsp

**Pull request merged**
